services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -176,9 +197,32 @@ async def upload_outputs(
                     logger.debug("No file %s to fetch port values from", data_file)
 
         if archiving_tasks:
-            await logged_gather(*archiving_tasks)
+            # NOTE: if one archiving task fails/cancelled all the ports are affected
+            # setting all other ports as finished with error/cancelled
+            try:
+                await logged_gather(*archiving_tasks)
```

Comment:
1. please use limited_gather.
Also there is an asyncGenerator available.

2. Why not create a specific function that does all this stuff instead of that weird thing? you know that if an exception is raised in one of them coroutine it will not stop the others? Please double check these things...